### ISSUE TITLE
Issue 1. Objects from view only visible mode missing elements. Issue 2. Rebar attributes use first value by default

### ISSUE THREAD

**Issue by dandad36** [LetsConstructIT/Grasshopper-TeklaDrawingLink#11]
I found two issues relating to reinforcement in drawing but maybe they apply for other elements too. I can't post the model so I took some screenshots of the issue. I have this section view. 
![image](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/assets/41262390/5e789ec8-16c9-49ca-92d0-ba21fb451730)

I'm using the Objects from view definition to get the drawing reinforcements in this view. 
![image](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/assets/41262390/cbf9ca0e-a265-4819-b49c-3f132b6e258b)

The All objects mode worked as expected and got all the reinforcements (changed to red lines in the pic)
![image](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/assets/41262390/0449000c-8f78-42b2-b714-6ddcdcdea332)

And when I used Only Visible mode, it's missing 1 reinforcement element (1 is still green)
![image](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/assets/41262390/33cbac1f-73e8-41e0-bd88-c6709d53fa77)

I've tested with some other similar section views and I got the same issue. One view couldn't find any reinforcement with only visibility mode despite being visible on the view. 

Another issue is the rebar attribute use the first value by default if there's no input. I have not locally defined any values. I was expecting the attributes to only change according to the input and any attributes without inputs would remain the same. Is this by design perhaps?
![image](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/assets/41262390/39c6092d-fb23-4ed7-b9ab-06879499414c)
![image](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/assets/41262390/0624bb75-2437-449e-ae2d-b50392652f34)
![image](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/assets/41262390/58faf3f0-39c2-4d8f-8ac4-6e195d3155ce)
![image](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/assets/41262390/746bfd94-d45d-4336-ad68-845176a8dec5)

**Comment by olszewski-grzegorz** [comment]
Hi! Thanks for your activity :) Screenshots are very helpful.
For future - please split the raised issues into separate ones. It will speed up things if they are separated.

For those two:
1) Objects from the view component. You're right, there was a bug related to the transformation of the section restriction box. I have addressed this issue in the newest release [2.3.1](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/releases/tag/2.3.1).
2) 'By design' is too strong word. I thought that users would mainly interact with this component based on pre-saved settings (standard Tekla attribute) which name can be provided by _Attributes Name_ input. Then, they would occasionally tweak some settings with other inputs.
This component is reading at the beginning default Tekla Rebar Attributes (with  Visibility set to All, Straight End symbol to No symbol and Hooked End symbol to No hook). After this, it loads the attribute file (if specified) and later overrides it with other inputs (if specified).

Your way of thinking sounds logical but the current component represents a different algorithm. It is consistent with the other attributes so I don't want to change it - too much involved time.

**Comment by dandad36** [comment]
Hi again, the issue of only visible objects is still there. I used another view and I compared the two modes in this picture. The non-drawing objects are missing. I think it's because model objects only intersecting the view restriction box and not completely inside it?
![image](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/assets/41262390/68645471-0c0b-498e-95dc-456f170e7e77)

![image](https://github.com/LetsConstructIT/Grasshopper-TeklaDrawingLink/assets/41262390/ec51c29c-a2b2-4a30-b34f-30b1945cd22a)

I managed to filter them myself using restriction box from the view property but it would be nice to have them directly from this definition.
